src/main/java/org/junitpioneer/jupiter/PioneerUtils.java

```diff
@@ -0,0 +1,164 @@
+
+/**
+ * Pioneer-internal utility class.
+ */
+class PioneerUtils {
+
+	private PioneerUtils() {
+		// private constructor to prevent instantiation of utility class
+	}
+
+	/**
+	 * Determines whether an annotation of any of the specified {@code annotationTypes}
+	 * is either <em>present</em>, <em>indirectly present</em>, or <em>meta-present</em>
+	 * on the test element (method or class) or any enclosing class belonging to the
+	 * specified {@code context}.
+	 */
+	public static boolean annotationsPresent(ExtensionContext context, Class<? extends Annotation>... annotationTypes) {
+		return Stream
+				.of(annotationTypes)
+				.map(annotationType -> findClosestAnnotation(context, annotationType))
+				.anyMatch(Optional::isPresent);
+	}
+
+	/**
+	 * Determines whether an annotation of any of the specified repeatable
+	 * {@code annotationTypes} is either <em>present</em>, <em>indirectly present</em>,
+	 * or <em>meta-present</em> on the test element (method or class) or any enclosing
+	 * class belonging to the specified {@code context}.
+	 */
+	public static boolean repeatableAnnotationsPresent(ExtensionContext context,
+			Class<? extends Annotation>... annotationTypes) {
+		return Stream
+				.of(annotationTypes)
+				.flatMap(annotationType -> findClosestRepeatableAnnotation(context, annotationType))
+				.iterator()
+				.hasNext();
+	}
+
+	/**
+	 * Returns the specified annotation if it is either <em>present</em>, <em>indirectly present</em>,
+	 * or <em>meta-present</em> on the test element (method or class) or any enclosing class belonging
+	 * to the specified {@code context}. If the annotation is present on more than one enclosing type,
+	 * the closest one is returned.
+	 */
+	public static <A extends Annotation> Optional<A> findClosestAnnotation(ExtensionContext context,
+			Class<A> annotationType) {
+		return findOnAnything(context, annotationType, false, false).findFirst();
+	}
+
+	/**
+	 * Returns the specified annotation if it is either <em>present</em>, <em>indirectly present</em>,
+	 * or <em>meta-present</em> on the test element (method or class) or any enclosing class belonging
+	 * to the specified {@code context}. If the annotation is present on more than one enclosing type,
+	 * all instances are returned.
+	 */
+	public static <A extends Annotation> Stream<A> findAllAnnotations(ExtensionContext context,
+			Class<A> annotationType) {
+		return findOnAnything(context, annotationType, false, true);
+	}
+
+	/**
+	 * Returns the specified repeatable annotation if it is either <em>present</em>, <em>indirectly present</em>,
+	 * or <em>meta-present</em> on the test element (method or class) or any enclosing class belonging
+	 * to the specified {@code context}. If the annotation is present on more than one enclosing type,
+	 * the instances on the closest one are returned.
+	 */
+	public static <A extends Annotation> Stream<A> findClosestRepeatableAnnotation(ExtensionContext context,
```

Comment:
Quoting the JavaDoc of, e.g., `AnnotationSupport::findAnnotation`:

> Find the first annotation of annotationType [...]

It seems that the word _annotation_ is used to describe the instance of an annotation type. Since that also keeps the name shorter, I'll be going with that interpretation.